webapi/src/main/scala/org/knora/webapi/routing/admin/ProjectsRouteZ.scala

```diff
@@ -82,11 +86,21 @@ final case class ProjectsRouteZ(
       projectCreateResponse <- projectsService.createProjectADMRequest(payload, requestingUser)
     } yield Response.json(projectCreateResponse.toJsValue.toString)
 
-  private def deleteProject(iriUrlEncoded: String, requestingUser: UserADM) =
+  private def deleteProject(iriUrlEncoded: String, requestingUser: UserADM): Task[Response] =
     for {
       iriDecoded            <- RouteUtilZ.urlDecode(iriUrlEncoded, s"Failed to URL decode IRI parameter $iriUrlEncoded.")
-      projectDeleteResponse <- projectsService.deleteProject(iriDecoded, requestingUser)
+      projectIri            <- Iri.ProjectIri.make(iriDecoded).toZIO
+      projectDeleteResponse <- projectsService.deleteProject(projectIri, requestingUser)
     } yield Response.json(projectDeleteResponse.toJsValue.toString())
+
+  private def changeProject(iriUrlEncoded: String, request: Request, requestingUser: UserADM): Task[Response] =
+    for {
+      iriDecoded            <- RouteUtilZ.urlDecode(iriUrlEncoded, s"Failed to URL decode IRI parameter $iriUrlEncoded.")
+      projectIri            <- Iri.ProjectIri.make(iriDecoded).toZIO
```

Comment:
The `make` function returns `Validation[ValidationException, ProjectIri]`. Shouldn't this error be mapped to a `BadRequestException` - here and above in the `deleteProject` case?

As far as I can see these cases are not tested, are they?

Reply:
Absolutely.
Regarding tests, you are right, until now we don't have negative tests at all for the ported routes. I will add them.

Reply:
Aha, I've just realized that the methods that would have to be tested in order to test these unhappy cases are private. The `ProjectsRouteZSpec` tests the `ProjectService` call which is inside these private methods. @seakayone do you still think these cases need to be tested? I'm obviously still unsure about the unit of unit tests..

Reply:
I added negative tests for updating projects. I will add them for the other use cases (routes), too. But in a separate PR.